src/main/java/core/basesyntax/service/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int OPERATION_INDEX = 0;
+    private static final int PRODUCT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int LINES_TO_SKIP = 1;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        if (inputReport == null || inputReport.isEmpty()) {
+            throw new IllegalArgumentException("Input report cannot be null or empty");
+        }
+
+        return inputReport.stream()
+                .skip(LINES_TO_SKIP)
+                .map(this::parseLine)
+                .collect(Collectors.toList());
+    }
+
+    private FruitTransaction parseLine(String line) {
```

Comment:
The method parseLine has to much responsibility = does to many things. SOLID says that calsses and methods should have one responsibility.
Additionaly parseLine name does not say precisely what it is doing. It is returning FruitTransaction object but parseLine name suggests it's transforming a String.